packages/x-charts/src/internals/plugins/featurePlugins/useChartHighlight/createIsHighlighted.ts

```diff
@@ -1,10 +1,20 @@
 import { HighlightScope } from './highlightConfig.types';
 import { HighlightItemData } from './useChartHighlight.types';
 
-export const createIsHighlighted =
-  (highlightScope: HighlightScope | null | undefined, highlightedItem: HighlightItemData | null) =>
-  (item: HighlightItemData | null): boolean => {
-    if (!highlightScope || !highlightedItem || !item) {
+function alwaysFalse(): boolean {
+  return false;
+}
+
+export function createIsHighlighted(
+  highlightScope: HighlightScope | null | undefined,
+  highlightedItem: HighlightItemData | null,
+) {
+  if (!highlightScope || !highlightedItem) {
+    return alwaysFalse;
+  }
```

Comment:
This is the important part. Now if `highlightScope` or `highlightedItem` aren't defined, we always return the same function, so no re-renders happen.